### PR TITLE
Remove unneeded gold files from tests

### LINKED ISSUE #97: Remove unnecessary gold files
## Description

There are some tests, particularly for the fixed-point iteration, which store data that can be quickly run during the CI. This should be removed.

## Context

Improving test quality

## Outputs

- Reference case runs prior to main tests
- These tests may also highlight if there are regression issues in other parts of the code, so these should also be reimplemented elsewhere.

### PR DESCRIPTION
**Pull request by mattfalcone1997**
## Summary

Remove gold files everywhere there isn't a regression test that needs an explicitly test with previous results

## Related Issue

Resolves #97 

## Checklist

- [x] Tests have been written for the new/changed behaviour.
- [x] Documentation/examples have been added/updated for the new changes.

### CODE REVIEW

File: test/tests/kernels/simple_diffusion/test.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class TestDiffusionProblem(unittest.TestCase):
+    """Test that simple kernels work within Hippo"""
+
+    def test_analytical_solution(self):
+        """Test diffusion kernel against analytical solution"""
+        coords, u = read_moose_exodus_data(RUN_DIR / "simple_diffusion_out.e", 1.0, "u")
+
+        assert all(abs(coords["x"] - u) < 3e-8), (
+            f"Max diff: {abs(coords['x'] - u).max()}"
```
Comment (k-collie) [inline test/tests/kernels/simple_diffusion/test.py:23]: It took me a minute to check that `u_analytic(x) = x` - maybe we should make a `u_analytic` function to make this explicit.

File: test/tests/postprocessors/side_advective_flux_integral/foam/constant/physicalProperties [hunk 2]
```diff
@@ -7,46 +7,42 @@
...
+    kappa 1; // Thermal conductivity [W/(m·K)]
+    mu 1.;
+  }
+  equationOfState
+  {
+    rho 0.5; // Density [kg/m^3]
+  }
 }
 
-
```
Comment (k-collie) [inline test/tests/postprocessors/side_advective_flux_integral/foam/constant/physicalProperties:51]: This file is now formatted differently to every other `physicalProperties`.

File: test/tests/postprocessors/side_advective_flux_integral/tests [hunk 3]
```diff
@@ -4,22 +4,30 @@
             type = RunCommand
             command = 'bash -c "foamCleanCase -case foam && blockMesh -case foam && decomposePar -case foam"'
         []
-        [check_csv]
-            type = CSVDiff
```
Comment (k-collie) [inline test/tests/postprocessors/side_advective_flux_integral/tests:8]: I'm not sure the CSV lines we save are worth introducing the python tests - maybe the python tests are more explicit about the structure of what we expect which is an improvement.

File: test/tests/problems/foam_problem/copies_values_from_openfoam_boundary/README [hunk 4]
```diff
@@ -1 +0,0 @@
-Copied from the OpenFOAM examples (with minor changes)
```
Comment (k-collie) [inline test/tests/problems/foam_problem/copies_values_from_openfoam_boundary/README:1]: Why do you want to delete these?

**Review by k-collie — COMMENTED** [review]
Looks good overall, a few comments for a change.

File: test/tests/postprocessors/side_advective_flux_integral/tests (hunk 3 above)
Comment (mattfalcone1997) [reply test/tests/postprocessors/side_advective_flux_integral/tests:8]: I want to make the tests explicit about the structure of the results

File: test/tests/postprocessors/side_advective_flux_integral/foam/constant/physicalProperties (hunk 2 above)
Comment (mattfalcone1997) [reply test/tests/postprocessors/side_advective_flux_integral/foam/constant/physicalProperties:51]: I think this an autoformat issue with my editor thinking the file is C++

File: test/tests/problems/foam_problem/copies_values_from_openfoam_boundary/README (hunk 4 above)
Comment (mattfalcone1997) [reply test/tests/problems/foam_problem/copies_values_from_openfoam_boundary/README:1]: Because they are no longer needed as the copying/imposition of boundary conditions is now done by `FoamVariable` and `FoamBC`. I simply added parallel versions of those tests where appropriate

**Review by k-collie — APPROVED** [review]
Perfect, thanks Matt!
